openmc/material.py

```diff
@@ -115,6 +115,30 @@ def __repr__(self):
 
         return string
 
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        if existing is None:
+            # If this is the first time we have tried to copy this object, create a copy
+            clone = type(self).__new__(type(self))
+            clone._id = self._id
+            clone._name = self._name
+            clone._density = self._density
+            clone._density_units = self._density_units
+            clone._nuclides = deepcopy(self._nuclides, memo)
+            clone._elements = deepcopy(self._elements, memo)
+            clone._sab = deepcopy(self._sab, memo)
+            clone._convert_to_distrib_comps = self._convert_to_distrib_comps
+            clone._distrib_otf_file = self._distrib_otf_file
```

Comment:
Ok, I think it is safe to also deepcopy it here as a stub for DD.
On Oct 30, 2015 9:09 AM, "Paul Romano" <notifications@github.com> wrote:

> In openmc/material.py
> <https://github.com/mit-crpg/openmc/pull/490#discussion_r43499774>:
>
> > @@ -115,6 +115,30 @@ def __repr__(self):
> >
> >          return string
> >
> > +    def __deepcopy__(self, memo):
> > +        existing = memo.get(id(self))
> > +
> > +        if existing is None:
> > +            # If this is the first time we have tried to copy this object, create a copy
> > +            clone = type(self).__new__(type(self))
> > +            clone._id = self._id
> > +            clone._name = self._name
> > +            clone._density = self._density
> > +            clone._density_units = self._density_units
> > +            clone._nuclides = deepcopy(self._nuclides, memo)
> > +            clone._elements = deepcopy(self._elements, memo)
> > +            clone._sab = deepcopy(self._sab, memo)
> > +            clone._convert_to_distrib_comps = self._convert_to_distrib_comps
> > +            clone._distrib_otf_file = self._distrib_otf_file
>
> I don't actually know; it's defined in *init* which is why I included it
> here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/490/files#r43499774>.
>